packages/connect/src/router-http-client.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright 2021-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import type { ConnectRouter } from "./router.js";
+import type { UniversalClientFn } from "./protocol/universal.js";
+import type { UniversalHandler } from "./protocol/universal-handler.js";
+import { ConnectError } from "./connect-error.js";
+import { createAsyncIterable } from "./protocol/async-iterable.js";
+import { Code } from "./code.js";
+
+/**
+ * An in-memory {@link UniversalClientFn | http client} that can be used to route requests to a {@link ConnectRouter}
+ * by passing network calls. Useful for testing and calling in-process services.
+ *
+ * @param router The {@link ConnectRouter} to use.
+ * @returns The {@link UniversalClientFn} that can be passed to `createTransport` functions.
+ */
+export function createRouterHttpClient(
+  router: ConnectRouter
+): UniversalClientFn {
```

Comment:
This is a nice function - doesn't do too much, but enables cool features 👍 

But I think we should still have a function on top that returns a `Transport`. This will be easier to explain,  gives us the leeway to add features (e.g. skip serialization, like we discussed), and doesn't require the user to import from @bufbuild/connect/protocol-* (which I would like to discourage).

WDYT about accepting `handlers: UniversalHandler[]` instead of `router: ConnectRouter` and moving it to `src/protocol/router-client.ts`? That way, we keep the exports from @bufbuild/connect to a minimum.

Reply:
I'll move it to protocol, I had the router in mind so I thought it was not possible but just using handlers makes sense to me.

I also wanted a Transport but was unsure where it will go as it will be tied to the protocol in its current form. Should we make an entirely new transport for this one or just use connect?

Reply:
I agree that it is odd to pick one of the protocols, but picking Connect seems reasonable. We have to create the ConnectRouter in the transport constructor function though for this to work, and cannot accept an existing ConnectRouter because createConnectRouter() takes options to enable/disable protocols.

I'm fine with adding a function that creates the Transport in this PR, or in a follow-up.